### PR TITLE
Doc update - array types

### PR DESCRIPTION
**Pull request by therealsilence**
- Add type-specific documentation on array types
- Update "forms vs tags" background doc to account for arrays

### CODE REVIEW

File: docs/synapse/userguides/storm_ref_type_specific.ipynb [hunk 1]
```diff
@@ -67,6 +68,279 @@
...
+    "print(q + q2)\n",
+    "# Execute the query to test it and get the packed nodes (podes).\n",
+    "podes = await core.eval(q1 + q2, num=1, cmdr=False)"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "Not all comparison operators are supported as part of the array \"by\" syntax. For example, equals ( ``=`` ) and prefix ( ``^=`` ) operators are supported, but other standard comparison operators (e.g., ``>=`` ) and extended comparison operators (e.g., ``~=`` ) are not.\n",
```
Comment (vEpiphyte) [inline docs/synapse/userguides/storm_ref_type_specific.ipynb]: This operator limitation sounds like a bug?

File: docs/synapse/userguides/storm_ref_type_specific.ipynb [hunk 2]
```diff
@@ -67,6 +68,279 @@
...
+    "print(q + q2)\n",
+    "# Execute the query to test it and get the packed nodes (podes).\n",
+    "podes = await core.eval(q1 + q2, num=1, cmdr=False)"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {},
+   "source": [
+    "Attempting to pivot from a typed property (primary or secondary) to an explicit array property of the same type (or vice versa) will generate an error. In other words, ``ou:name=\"Foo Corporation\" -> ou:org:names`` is an invalid pivot.\n",
```
Comment (vEpiphyte) [inline docs/synapse/userguides/storm_ref_type_specific.ipynb]: This also smells like a bug

File: docs/synapse/userguides/storm_ref_type_specific.ipynb (hunk 1 above)
Comment (therealsilence) [reply docs/synapse/userguides/storm_ref_type_specific.ipynb]: WM-372

File: docs/synapse/userguides/storm_ref_type_specific.ipynb (hunk 2 above)
Comment (therealsilence) [reply docs/synapse/userguides/storm_ref_type_specific.ipynb]: WM-374

**Pull request merged**